### PR TITLE
refactor: remove wrapUrlsInBlankATag in favour of transformDotsInUrlsInvisibly

### PR DESCRIPTION
**Pull request by stefro**
## Summary
https://app.clickup.com/t/1rgq58b

Follow up of https://github.com/ArkEcosystem/marketsquare.io/pull/2319 . Although the tags work for most email clients, the MS outlook app ignores it and still turns it into a link.

I had to completely change the approach since wrapping the string in an empty a tag and try to encode the `.` in some sort of way didn't seem to help. I finally figured that adding an extra `.` would stop the email clients from parsing the strings as urls. By hiding the extra `.` it won't look weird to the user.
This is even better as the first approach, since we don't need to add styling anymore to make it look like a normal string.

This new Macro will replace the `Str::macro('wrapUrlsInBlankATag')` since the naming of this Macro does not describe the way we handle it now.
As soon as this old Macro is not being used anymore, we can safely remove it from the codebase.

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I provided a screenshot of my changes to the component _(if applicable)_
-   [ ] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [ ] I added an explanation on how to use the component to the readme _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

File: src/Providers/StringMacroServiceProvider.php [hunk 1]
```diff
@@ -16,11 +17,17 @@ public function register(): void
 
     public function boot(): void
     {
+        // This macro is depreciated in favor of Str::transformUrlsInvisibly().
+        // As soon as it is not being used anymore, it can be deleted.
```
Comment (ItsANameToo) [inline src/Providers/StringMacroServiceProvider.php:21]: you can already remove it in this PR, it's only used on marketsquare and will be replaced by the new one once this change becomes available

File: src/Providers/StringMacroServiceProvider.php [hunk 2]
```diff
@@ -16,11 +17,17 @@ public function register(): void
...
         Str::macro('wrapUrlsInBlankATag', function ($str, $attributes = []): string {
             $wrapper = new UrlInStringWrapper($str);
             $wrapper->setAttributes($attributes);
 
             return $wrapper->getString();
         });
+
+        Str::macro('transformUrlsInvisibly', function ($str): string {
```
Comment (ItsANameToo) [inline src/Providers/StringMacroServiceProvider.php:29]: the name does not really explain me what it does 🤔 maybe something like `transformDotsInUrls` or `transformDotsInUrlsInvisibly` would better explain its purpose at a glance?

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

File: src/Providers/StringMacroServiceProvider.php (hunk 2 above)
Comment (stefro) [reply src/Providers/StringMacroServiceProvider.php]: I like `transformDotsInUrlsInvisibly ` 👍  I've changed it!

File: src/Providers/StringMacroServiceProvider.php (hunk 1 above)
Comment (stefro) [reply src/Providers/StringMacroServiceProvider.php]: done

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**